src/coreclr/jit/instr.cpp

```diff
@@ -758,7 +758,7 @@ void CodeGen::inst_RV_SH(
 //    This method is not idempotent - it can only be called once for a
 //    given node.
 //
-CodeGen::OperandDesc CodeGen::genOperandDesc(GenTree* op)
+CodeGen::OperandDesc CodeGen::genOperandDesc(GenTree* op, insOpts instOptions, var_types simdBaseType)
```

Comment:
Do we need `simdBaseType`? Couldn't it be derived from `op` or one of its children? E.g., shouldn't a contained VecCon node have the correct type? If that's true, then presumably we wouldn't need `insOpts` either, since the types should already be correct.

Reply:
passing `simdBaseType` is mainly for emitting the constant vector, `VecCon` node only specifies the simd length as its type, e.g. `TYP_SIMD32`. When embedded broadcast enabled, we want emit a constant scalar based on the base type of the vector, so we may need this information here.

Reply:
What does the IR look like in this case? If someone determines we should use embedded broadcast, shouldn't the VecCon node be replace by an appropriate scalar node (GT_CNS_INT, say?) which would have the right size?

Reply:
Based on the previous [discussion](https://github.com/dotnet/runtime/pull/84821#issuecomment-1508583897), I generally follow the idea that keep the `VecCon` as it is, and emit the scalar if embedded broadcast can be enabled. On the other hand, the replacement from Vector to scalar can be down in the earlier stage indeed, I guess we shall discuss which design can be better.

Reply:
I thought based on a later discussion (https://github.com/dotnet/runtime/pull/84821#discussion_r1185371882) we determined that lowering the `VecCon` to a `BroadcastScalarToVector` would be suitable and avoid this?

That is, we want to have a `VecCon` up through lowering if a `VecCon` is possible. This allows the most optimizations in the form of `constant folding`, `CSE`, etc

When we get to lowering, we can then opt to change the `VecCon` into a contained `Broadcast` if that's viable. This then allows us to centralize the logic to only handling `Broadcast` including the constant emission logic/etc.

Perhaps there was some miscommunication on my end, however?

Reply:
Ah, I was thinking we were following this [method](https://github.com/dotnet/runtime/pull/84821#discussion_r1186351484) to avoid unfolding the node, but your point here makes sense to me, I will try to do the transform from `VecCon` to `Broadcast` tomorrow.

Reply:
Ah, I think I misread the initial comment but see now what you were asking. That's my fault for the confusion then

Reply:
What kind of a node is `Broadcast` then? Would it have a scalar argument and produce a vector result, but be contained and thus cause generating the embedded broadcast instruction?

e.g., `ADD[SIMD64<int>](LCL_VAR[SIMD64<int>], Broadcast[SIMD64<int>](GT_CNT_INT[INT])` ?

(where `[ ]` denotes node type)

Reply:
Broadcast is an existing hwintrinsic node that we already need to check and handle as part of this optimization, so it’s a natural node to use if we want to make things consistent.

it takes either a scalar value or an address and produces a vector, so we can contain it if it’s input is itself contained (either a constant or coming from memory)